### PR TITLE
Add CNP and K8s NP equalness and missing func

### PR DESCRIPTION
**Pull request by aanm**
Read per commit basis.

```release-note
Stop triggering unnecessary policy regenerations for every Kubernetes watch event.
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5519)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-missed-k8s

**Comment by aanm** [comment]
test-upstream-k8s

**Comment by ianvernon** [comment]
@aanm no tests have ran for this PR, as you are merging it into another branch, which I guess the CI infrastructure doesn't support. I will leave this to you to merge based off of approvals of relevant CODEOWNERs into your PR.

File: pkg/k8s/apis/cilium.io/utils/utils.go
Comment (ianvernon) [reply pkg/k8s/apis/cilium.io/utils/utils.go]: Add documentation and unit testing.

File: pkg/k8s/apis/cilium.io/v2/types.go
Comment (ianvernon) [inline pkg/k8s/apis/cilium.io/v2/types.go:200]: Also add unit tests here as well.

**Review by ianvernon — COMMENTED** [review]
Looks good, but needs a few more unit tests / some documentation is missing.

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**
